### PR TITLE
updates for deleting attrs and rels from schema

### PR DESCRIPTION
**Pull request by ajtmccarty**
IFC-842

our logic to delete an attribute or relationship from a schema would only delete the `SchemaNode` -> `SchemaAttribute`/`SchemaRelationship` relationship without actually deleting the `SchemaAttribute`/`SchemaRelationship`. this is illegal according to our internal schema b/c the `node` relationship of a `SchemaAttribute`/`SchemaRelationship` is a mandatory, cardinality-one relationship.

this would not cause any issues for end-users, but our schema validation logic would identify the problem and raise an error, which I learned while trying to update our rebase and merge logic to do more validation

### CODE REVIEW

File: backend/tests/integration/schema_lifecycle/shared.py [hunk 1]
```diff
@@ -35,7 +35,7 @@ def schema_person_02_first_last(self, schema_person_base) -> Dict[str, Any]:
         """Rename the attribute name to firstname and add a new lastname attribute."""
         assert schema_person_base["attributes"][0]["name"] == "name"
         schema_person_base["attributes"][0]["name"] = "firstname"
-        schema_person_base["attributes"].append({"name": "lastname", "kind": "Text"})
+        schema_person_base["attributes"].append({"name": "lastname", "kind": "Text", "optional": True})
```
Comment (ajtmccarty) [inline backend/tests/integration/schema_lifecycle/shared.py:38]: pretty sure this needs to be optional or have a default

File: backend/infrahub/core/schema/manager.py [hunk 2]
```diff
@@ -389,82 +389,113 @@ async def update_node_in_db_based_on_diff(  # pylint: disable=too-many-branches,
...
+
+        item_ids = set()
+        item_names = set()
+        for field in node.local_attributes + node.local_relationships:
+            if field.name not in attrs_rels_to_update:
+                continue
+            if field.id:
+                item_ids.add(field.id)
+                item_names.add(field.name)
+        missing_field_names = list(attrs_rels_to_update - item_names)
```
Comment (ajtmccarty) [inline backend/infrahub/core/schema/manager.py:412]: attributes and relationships that are removed on this schema will already be removed from the schema at this point, so we need to check if there are items in the diff that are not on the node

File: backend/infrahub/core/schema/manager.py [hunk 3]
```diff
@@ -389,82 +389,113 @@ async def update_node_in_db_based_on_diff(  # pylint: disable=too-many-branches,
...
+                include_source=True,
+            )
+            missing_rels = await self.query(
+                db=db,
+                branch=branch,
+                schema=relationship_schema,
+                filters={"name__values": missing_field_names, "node__id": node.id},
+                include_owner=True,
+                include_source=True,
+            )
```
Comment (ajtmccarty) [inline backend/infrahub/core/schema/manager.py:439]: then we get them from the database if we need to

**Review by dgarros — APPROVED** [review]
LGTM as a quick fix but it feels like the right long term solution would be to add a `on_delete:Cascade` to the internal schema for the relationships between SchemaNode and SchemaAttribute / SchemaRelationship

For it to be effective we'll also need to update the method that update the schema to used`NoeManager.delete()`

**Pull request merged**